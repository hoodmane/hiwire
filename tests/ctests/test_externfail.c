#include "testlib.h"

void
hiwire_invalid_id(int type, HwRef ref)
{
  printf("type: %d, ref: %d\n", type, ref);
}

int
main()
{
  hiwire_get(NULL);
  HwRef id1 = hiwire_new(int_to_ref(5));
  hiwire_decref(id1);
  hiwire_get(id1);
  hiwire_incref(id1);
  hiwire_decref(id1);

  return 0;
}
