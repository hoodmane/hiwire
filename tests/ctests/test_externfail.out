type: 1, ref: 0
type: 1, ref: 3
type: 2, ref: 3
type: 3, ref: 3
